beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -48,19 +48,41 @@ func (vs *Server) setExecutionData(ctx context.Context, blk interfaces.SignedBea
 	if err != nil {
 		log.WithError(err).Warn("Proposer: failed to check if builder can be used")
 	} else if canUseBuilder {
-		h, err := vs.getPayloadHeaderFromBuilder(ctx, slot, idx)
+		builderPayload, err := vs.getPayloadHeaderFromBuilder(ctx, slot, idx)
 		if err != nil {
 			builderGetPayloadMissCount.Inc()
 			log.WithError(err).Warn("Proposer: failed to get payload header from builder")
 		} else {
-			blk.SetBlinded(true)
-			if err := blk.SetExecution(h); err != nil {
-				log.WithError(err).Warn("Proposer: failed to set execution payload")
+			// Can only check values if block is Capella or later.
```

Comment:
This logic is complex and deeply nested. Would recommend splitting up into a smaller helper and testing it independently

Reply:
I used a switch case instead of if else. Other than that, it's really hard to split this function up. The concept of it is fairly straightforward. All the local block failures, return. All the remote block failures. cont to use local block